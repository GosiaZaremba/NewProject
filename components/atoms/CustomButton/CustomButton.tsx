import React from 'react';
import {Button} from 'react-native';

export type Props = {
  title: string;
  onPress: () => void;
  color: string;
};
export const CustomButton: React.FC<Props> = ({title, onPress, color}) => {
  return <Button title={title} onPress={onPress} color={color}></Button>;
};
