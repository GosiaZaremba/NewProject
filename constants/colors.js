export const Colors = {
  pink: '#fc296f',
  red: '#e63039',
  navy: '#39539b',
  blue: '#3872c3',
  grey: '#908D8E',
  light: '#F5F0F2',
};
