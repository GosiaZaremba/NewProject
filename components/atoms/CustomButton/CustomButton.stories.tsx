import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { CustomButton } from './CustomButton';
import { Colors } from '../../../constants/colors';

storiesOf('atoms/CustomButton', module)
    .addDecorator((story) => story())
    .add('default', () => (
        <CustomButton
            title={'Button title'}
            color={Colors.pink}
            onPress={() => {}}
        />
    ));

// export default {
//   title: 'components/atoms/CustomButton/CustomButton',
//   component: CustomButton,
// } as ComponentMeta<typeof CustomButton>;

// export const Basic: ComponentStory<typeof CustomButton> = args => (
//   <CustomButton {...args} />
// );

// Basic.args = {
//     title: 'Button title',
//     color: Colors.pink,
//     accessibilityLabel: 'Button title',
// };
