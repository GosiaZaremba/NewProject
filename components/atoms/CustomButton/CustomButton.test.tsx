import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { CustomButton } from './CustomButton';
import { Colors } from '../../../constants/colors';

describe('Button Atom', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <CustomButton
                    title={'Button title'}
                    color={Colors.pink}
                    onPress={() => {}}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <CustomButton
                title={'Button title'}
                color={Colors.pink}
                onPress={() => {}}
                testID="atom-button"
            />
        );
        const customButton = getByTestId('atom-button');
        expect(customButton).toHaveTextContent('Button title');
    });
});
