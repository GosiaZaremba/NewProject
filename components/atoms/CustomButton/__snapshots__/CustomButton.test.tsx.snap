// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom Button Component Snapshot 1`] = `
<View
  accessibilityLabel="Button title"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="atom-button"
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#fc296f",
          },
        ]
      }
    >
      Button title
    </Text>
  </View>
</View>
`;
